modules/social_features/social_event/src/Entity/Node/Event.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace Drupal\social_event\Entity\Node;
+
+use Drupal\social_event\EventEnrollmentInterface;
+use Drupal\social_node\Entity\Node as Node;
+
+/**
+ * Defines bundle class for event node type.
+ */
+final class Event extends Node implements EventInterface {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function isEnrollmentEnabled(): bool {
+    // Get global event settings.
+    $settings = \Drupal::config('social_event.settings');
+    if (
+      $settings->get('disable_event_enroll') ||
+      !$this->hasField('field_event_enroll') ||
+      (!$this->get('field_event_enroll')->isEmpty() && !$this->get('field_event_enroll')->getString())
+    ) {
+      return FALSE;
+    }
+
+    // When field was added to the event the value become `null` which
+    // creates some inconsistent if after saving the event the value is `false`.
+    $was_not_changed = $this->get('field_event_enroll')->isEmpty();
+    $is_enabled = (bool) $this->get('field_event_enroll')->getString();
```

Comment:
Why are we typecasting?

Reply:
Because `getString()` returns `1` or `0` for boolean field, but we need to have `bool` type value to make possible having "strict type" conditions.